x-pack/test/dataset_quality_api_integration/tests/data_streams/data_stream_details.spec.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { log, timerange } from '@kbn/apm-synthtrace-client';
+import expect from '@kbn/expect';
+import { DatasetQualityApiClientKey } from '../../common/config';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { expectToReject, getDataStreamSettingsOfFirstIndex } from '../../utils';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+  const synthtrace = getService('logSynthtraceEsClient');
+  const esClient = getService('es');
+  const datasetQualityApiClient = getService('datasetQualityApiClient');
+  const start = '2023-12-11T18:00:00.000Z';
+  const end = '2023-12-11T18:01:00.000Z';
+  const dataStream = 'nginx.access';
+
+  async function callApiAs(user: DatasetQualityApiClientKey, datasetQuery: string) {
+    return await datasetQualityApiClient[user]({
+      endpoint: 'GET /internal/dataset_quality/data_streams/details',
+      params: {
+        query: {
+          datasetQuery,
+        },
+      },
+    });
+  }
+
+  registry.when('DataStream Details', { config: 'basic' }, () => {
+    describe('gets the data stream details', () => {
+      before(async () => {
+        await synthtrace.index([
+          timerange(start, end)
+            .interval('1m')
+            .rate(1)
+            .generator((timestamp) =>
+              log
+                .create()
+                .message('This is a log message')
+                .timestamp(timestamp)
+                .dataset(dataStream)
+                .defaults({
+                  'log.file.path': '/my-service.log',
+                })
+            ),
+        ]);
+      });
+
+      it('returns error when dataset query arg is not provided', async () => {
+        expect(
+          (await expectToReject(() => callApiAs('datasetQualityLogsUser', ''))).message
+        ).to.contain('Dataset query cannot be empty');
+      });
+
+      it('returns 404 if matching data stream is not available', async () => {
```

Comment:
could we modify this test a bit?
My proposal: instead of checking the message length, we could directly check the error code as it's stated in the title of the test, we could do something like

```
      const err = await expectToReject(
        async () =>
          await callApiAs('datasetQualityLogsUser', nonExistentDataStream))
      );

      expect(err.res.status).to.be(404);
      expect(err.res.body.message).to.contain(expectedMessage);
```